### PR TITLE
fix: Ticket Order: User able to add invalid email 

### LINKED ISSUE #5277: Ticket Order: User able to add invalid email
A user could get a ticket and add an invalid email. Compare at https://eventyay.com/orders/d2b49133-ddc1-4f77-865e-f12cb8e24108/view

Expected:
* User should only be able to add valid emails
* Question: For the first ticket should we limit this and only allow the user to use their confirmed email for free tickets? How do other systems like eventbrite do it?

Screenshot from 2020-10-09 21-44-14

### PR DESCRIPTION
**Pull request by daretobedifferent18**
Fixes #5277 

#### Short description of what this resolves:
Email are only added if they are valid

#### Changes proposed in this pull request:

- Added the redExp for email

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by daretobedifferent18** [comment]
@iamareebjamal Please see the change I have made. And also I have one more suggestion, there are other places as well besides this where email validation is failing, so should I fix them also??

Screen Shot 2020-11-18 at 20 58 17  Screen Shot 2020-11-18 at 20 58 22
Screen Shot 2020-11-18 at 20 58 25  Screen Shot 2020-11-18 at 20 58 29
Screen Shot 2020-11-18 at 20 58 34  Screen Shot 2020-11-18 at 20 58 42

**Comment by iamareebjamal** [comment]
Failing for valid email `sad:jamal.areeb@mail.ab`

**Comment by daretobedifferent18** [comment]
> Failing for valid email `sad:jamal.areeb@mail.ab`

is it so?

**Comment by daretobedifferent18** [comment]
@iamareebjamal pls validate and see if its working now?

**Comment by daretobedifferent18** [comment]
@iamareebjamal ofcourse we can't validate every possible case b for that validating through OTP would be best but for now this sol looks good to me then before where invalid emails where not validated

**Comment by iamareebjamal** [comment]
Exactly. It's better to allow invalid emails than not allow valid emails

**Comment by daretobedifferent18** [comment]
> Exactly. It's better to allow invalid emails than not allow valid emails

So should I do now anything for it?

**Comment by iamareebjamal** [comment]
The regex should not filter out any valid email

**Comment by maze-runnar** [comment]
I think current validation is ok because adding any regex for validating accurate email address will surely filter out valid ones. Because the validation of email is also a part of the provider.  some emails are valid but not acceptable by one provider while acceptable by another provider. Even RFC can't validate 100% emails.

**Comment by iamareebjamal** [comment]
https://ihateregex.io/expr/email/

**Comment by daretobedifferent18** [comment]
@iamareebjamal 
> https://ihateregex.io/expr/email/
> 
> Simple enough and does not filter out most of valid emails.

OK
> there are other places as well besides this where email validation is failing, so should I fix them also??

?

**Comment by iamareebjamal** [comment]
No, just here

**Comment by daretobedifferent18** [comment]
> No, just here

ok I'll do it everywhere

**Comment by daretobedifferent18** [comment]
@iamareebjamal I have done the changes, and eslint error  error is coming.

**Comment by iamareebjamal** [comment]
yarn lint

**Comment by daretobedifferent18** [comment]
> yarn lint

how to fix this yarn lint error?

**Comment by iamareebjamal** [comment]
yarn linr

**Comment by daretobedifferent18** [comment]
> yarn lint

Screenshot from 2020-11-19 19-43-46
This is coming

**Comment by daretobedifferent18** [comment]
either we have to disable no-control-regex

File: app/utils/validators.js [hunk 1]
```diff
@@ -79,6 +79,10 @@ export const validPhoneNumber = new RegExp(
   '^\\s*(?:\\+?(\\d{1,3}))?([-. (]*(\\d{3})[-. )]*)?((\\d{3})[-. ]*(\\d{2,4})(?:[-.x ]*(\\d+))?)\\s*$'
 );
 
+export const validEmail = new RegExp(
+  '[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+'
```
Comment (iamareebjamal) [inline app/utils/validators.js:83]: Disable eslint `no-control-regex` error on this line

Comment (daretobedifferent18) [reply app/utils/validators.js:84]: Done

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**